File: src/Web/src/app/core/models/messages/bot-message.ts
import {Message} from "@IQuality/core/models/messages/message";
import {Suggestion} from "@IQuality/core/models/suggestion";
import {ResponseType} from "@IQuality/core/models/messages/response-type";
import {Listable} from "@IQuality/core/models/listable";
import {GraphData} from "@IQuality/core/models/graph-data/graph-data";

export class BotMessage extends Message{
  id: string;

  chatId: string;
  sendDate: Date;
  content: string;

  listData?: Array<Listable>
  graphData? : GraphData

  responseType: ResponseType;

  suggestions: Array<Suggestion>;
}

File: src/Models/Chat/Messages/BotMessage.cs
using System.Collections.Generic;
using IQuality.Models.Dialogflow;
using IQuality.Models.Forms;
using IQuality.Models.Interfaces;

namespace IQuality.Models.Chat.Messages
{
    public class BotMessage : BaseMessage
    {
        public string Content { get; set; }
        public List<Suggestion> Suggestions { get; set; }
        public List<Listable> ListData { get; set; }
        public ResponseType ResponseType { get; set; }

        public BotMessage()
        {
            ResponseType = ResponseType.Text;
            ListData = new List<Listable>();
            Suggestions = new List<Suggestion>();
            Content = string.Empty;
        }

        public void RespondText(string content)
        {
            Content = content;
            ResponseType = ResponseType.Text;
        }

        public void RespondList(string content, List<Listable> data)
        {
            Content = content;
            ListData = data;
            ResponseType = ResponseType.List;
        }
    }
}